test/conformance/kuttl/generate/policy/cornercases/pol-clone-create-on-trigger-deletion/01-manifests.yaml

```diff
@@ -0,0 +1,47 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: pol-clone-create-on-trigger-deletion-manifest-ns
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: pol-clone-create-on-trigger-deletion-trigger
+  namespace: pol-clone-create-on-trigger-deletion-manifest-ns 
+---
+apiVersion: v1
+data:
+  foo: YmFy
+kind: Secret
+metadata:
+  name: pol-clone-create-on-trigger-deletion-secret
+  namespace: pol-clone-create-on-trigger-deletion-manifest-ns
+type: Opaque
+---
+apiVersion: kyverno.io/v2beta1
+kind: Policy
+metadata:
+  name: pol-clone-create-on-trigger-deletion
+  namespace: pol-clone-create-on-trigger-deletion-manifest-ns
+spec:
+  rules:
+  - name: clone-secret
+    match:
+      any:
+      - resources:
+          kinds:
+          - ConfigMap
+    preconditions:
+      any:
+      - key: "{{ request.operation }}"
+        operator: Equals
+        value: DELETE
+    generate:
+      apiVersion: v1
+      kind: Secret
+      name: regcred
```

Comment:
This isn't correct. The generated name should not be the same as the clone source name.

Reply:
@chipzoller i just corrected this but it fails, which shouldn't be. could this be an issue with the implementation?

Reply:
If you make all the changes and push them up, we'll see.

Reply:
@chipzoller this is the failing testcase, [generate/policy/cornercases/pol-clone-create-on-trigger-deletion](https://github.com/kyverno/kyverno/actions/runs/4429556220/jobs/7770312336#step:9:324)

Reply:
the source, `regcred` does exist but the generated resource `pol-clone-create-on-trigger-deletion-secret` was not found

Reply:
@realshuting can you take a look at this test case and understand why it would be failing? Everything looks good to me. Do we have a code problem somewhere? The Secret should be generated but it appears to not be.

Reply:
@realshuting have you been able to confirm this

Reply:
The test set looks good, couldn't find any obvious error in the background controller's log.

@yimikao - have you tested it locally and dump the admission controller's log? I want to check if the UR was created upon cm deletion.

Reply:
@yimikao - I've tested it locally and it worked fine for me:
```
# kubectl -n pol-clone-create-on-trigger-deletion-manifest-ns delete cm pol-clone-create-on-trigger-deletion-trigger
configmap "pol-clone-create-on-trigger-deletion-trigger" deleted

# kubectl -n pol-clone-create-on-trigger-deletion-manifest-ns get secret
NAME                                          TYPE     DATA   AGE
pol-clone-create-on-trigger-deletion-secret   Opaque   1      11s
```

You may want to check if there are any duplicate names used across the entire test suites. It could cause kuttl tests to fail if two tests have resources with same name.

Reply:
I typically use the name of the test case as the Namespace name. I'm also wondering if there is a UR that's in completed status here. Doesn't seem like there would be though.

Reply:
> @yimikao - I've tested it locally and it worked fine for me:
> 
> ```
> # kubectl -n pol-clone-create-on-trigger-deletion-manifest-ns delete cm pol-clone-create-on-trigger-deletion-trigger
> configmap "pol-clone-create-on-trigger-deletion-trigger" deleted
> 
> # kubectl -n pol-clone-create-on-trigger-deletion-manifest-ns get secret
> NAME                                          TYPE     DATA   AGE
> pol-clone-create-on-trigger-deletion-secret   Opaque   1      11s
> ```
> 
> You may want to check if there are any duplicate names used across the entire test suites. It could cause kuttl tests to fail if two tests have resources with same name.

alright, i'll check

Reply:
@realshuting works locally for me too
```
# kubectl get secrets -A

pol-clone-create-on-trigger-deletion-manifest-ns   pol-clone-create-on-trigger-deletion-secret               Opaque                          1      13s
pol-clone-create-on-trigger-deletion-manifest-ns   regcred                                                   Opaque                          1      65s
```

[resource name is also unique](https://pasteboard.co/0xxdIBOr7KGC.png)

Reply:
@realshuting i tried this again, same result